src/plugins/intel_npu/tools/protopipe/src/simulation/layers_reader.cpp

```diff
@@ -42,5 +53,9 @@ InOutLayers LayersReader::readLayers(const InferenceParams& params) {
     // NB: Using OpenVINO to read the i/o layers information for *.onnx model
     OpenVINOParams ov;
     ov.path = OpenVINOParams::ModelPath{ort.model_path, ""};
-    return getOVReader().readLayers(ov, true /* use_results_names */);
+    auto inOutLayers = getOVReader().readLayers(ov, true /* use_results_names */);
+    adjustLayerPartialShapes(inOutLayers.in_layers);
+    adjustLayerPartialShapes(inOutLayers.out_layers);
```

Comment:
I believe it doesn't make sense for `out_layers`. Does `onnxruntime_perf_test` do the same for output layers?

Reply:
Well I can't tell for sure but it looks like it: 
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/test/perftest/ort_test_session.cc#L938-L951
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/test/perftest/ort_test_session.cc#L1042-L1045

Reply:
make sense